src/test/deterministic.rs

```diff
@@ -1,32 +1,55 @@
-use crate::test::externalities::TestExternalities;
+use crate::test::externalities::{TestExternalities, TxPoolExtApi};
 use crate::{
 	node::InternalNode,
 	rpc::{self, RpcExtension},
 };
-use jsonrpc_core_client::transports::local;
 use jsonrpc_core_client::RpcChannel;
 use manual_seal::rpc::ManualSealClient;
+use sp_core::{
+	offchain::TransactionPoolExt,
+	testing::{KeyStore, SR25519},
+	traits::KeystoreExt,
+};
+use sp_externalities::{Externalities, ExternalitiesExt};
+use sp_keyring::sr25519::Keyring;
 use std::ops::{Deref, DerefMut};
 
 /// A deterministic internal instance of substrate node.
-pub struct Deterministic<Runtime> {
-	node: InternalNode<Runtime>,
+pub struct Deterministic<Runtime: frame_system::Trait> {
+	node: InternalNode,
+	externalities: TestExternalities<Runtime>,
 }
 
-impl<Runtime> rpc::RpcExtension for Deterministic<Runtime> {
+impl<Runtime: frame_system::Trait> rpc::RpcExtension for Deterministic<Runtime> {
 	fn rpc<TClient: From<RpcChannel> + 'static>(&mut self) -> TClient {
-		use futures01::Future;
-		let rpc_handler = self.node.rpc_handler();
-		let (client, fut) = local::connect::<TClient, _, _>(rpc_handler);
-		self.node.tokio_runtime().spawn(fut.map_err(|_| ()));
-
-		client
+		self.node.rpc_client()
 	}
 }
 
-impl<Runtime> Deterministic<Runtime> {
-	pub fn new(node: InternalNode<Runtime>) -> Self {
-		Self { node }
+impl<Runtime: frame_system::Trait> Deterministic<Runtime> {
+	pub fn new(node: InternalNode) -> Self {
```

Comment:
I kind of preferred when `InternalNode` had `Runtime` generic type as well so it was type checked. Now it seems you can create any kind of `InternalNode` and then attempt to use it as any `Runtime` when wrapped in `Deterministic`.

As discussed privately, we might want to introduce some trait that will bound `Runtime` and the types required to construct `InternalNode`: `Block, RuntimeApi, Executor, ChainSpecFactory` so that both are used together.

Fine with this approach for now though.